app/src/main/java/com/example/tkey_android/FirstFragment.java

```diff
@@ -737,155 +881,390 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
-                            Exception e = ((Result.Error<Boolean>) result).exception;
-                            renderError(e);
-                            hideLoading();
-                        });
-                    } else if (result instanceof Result.Success) {
-                        Boolean set = ((Result.Success<Boolean>) result).data;
-                        Snackbar snackbar = Snackbar.make(view1, "Set private key result: " + set, Snackbar.LENGTH_LONG);
-                        snackbar.show();
-                        hideLoading();
+                    boolean ready;
+                    try {
+                        ready = client.isReady();
+                    } catch (Exception e) {
+                        throw new EthereumSignerError(EthereumSignerError.ErrorType.UNKNOWN_ERROR);
                     }
-                });
-            } catch (RuntimeError e) {
-                renderError(e);
-                hideLoading();
-            }
-        });
 
-        binding.getPrivateKey.setOnClickListener(view1 -> {
-            showLoading();
-            try {
-                String key = PrivateKeysModule.getPrivateKeys(activity.tKey);
-                Snackbar snackbar = Snackbar.make(view1, key, Snackbar.LENGTH_LONG);
-                snackbar.show();
-                hideLoading();
-            } catch (RuntimeError e) {
-                renderError(e);
-                hideLoading();
-            }
-        });
+                    if (ready) {
+                        String msg = "hello world";
+                        String msgHash = TSSHelpers.hashMessage(msg);
+                        Triple<BigInteger, BigInteger, Byte> signatureResult;
+                        try {
+                            signatureResult = client.sign(msgHash, true, msg, precompute, signatureString);
+                        } catch (TSSClientError e) {
+                            throw new RuntimeException(e);
+                        }
+                        try {
+                            client.cleanup(signatureString.toArray(new String[0]));
+                        } catch (TSSClientError e) {
+                            throw new RuntimeException(e);
+                        }
 
-        binding.getAccounts.setOnClickListener(view1 -> {
-            showLoading();
-            try {
-                ArrayList<String> accounts = PrivateKeysModule.getPrivateKeyAccounts(activity.tKey);
-                Snackbar snackbar = Snackbar.make(view1, accounts.toString(), Snackbar.LENGTH_LONG);
-                snackbar.show();
-                hideLoading();
-            } catch (RuntimeError | JSONException e) {
-                renderError(e);
-                hideLoading();
-            }
-        });
+                        String uncompressedPubKey;
+                        try {
+                            uncompressedPubKey = new KeyPoint(pubKey.get()).getPublicKey(KeyPoint.PublicKeyEncoding.FullAddress);
```

Comment:
Reliance on keypoint here will need to be removed when moving this signing code to separate library

Reply:
The web3 provider for android must have not have tkey as a direct dependency, only data from tkey is provided to it.

See `public static ECPoint parsePublicKey(byte[] serializedKey)` in the tss-client sdk for android to decompress the key here.